### PR TITLE
Update carts for 23.11 template

### PR DESCRIPTION
**Pull request by omuzzyEBSCO**
#This update resolves issues with:
- download cart functionality with EDS items
- send basket functionality with EDS items
- fixes minor 23.11 issues/typos

##To test this update:
- Download the updated kpz here: [https://github.com/ebsco/edsapi-koha-plugin/blob/OM-cart-issues-2311-rework/eds_plugin_23.11001.kpz]
- Install the kpz in your existing Koha instance
- Be sure to click configure and if necessary, update your settings and save

##To test functionality of Download Cart:
- navigate to your library's Koha OPAC interface
- make sure that Discovery is selected in the search bar dropdown menu
- perform a search 
- Click 'Add to Cart' on an EDS item
- Click on your Cart in the top left corner
- Click Download in the dropdown menu, and select any download format
- Confirm that the file downloads in the format as expected
- Feel free to download multiple formats to test

##To test functionality of Send Basket:
- Ensure that your library's Koha instance is set up to send email
- navigate to your library's Koha OPAC interface
- make sure that Discovery is selected in the search bar dropdown menu
- perform a search 
- Click 'Add to Cart' on an EDS item
- Click on your Cart in the top left corner
- Click on Send
- Enter your email and proceed with sending item in the 'Sending your cart' dialogue box
- Check the email address to ensure that the message was received, with an ISO2709 attachment.

##Resolves the following issues:
#131 
#132

### CODE REVIEW

File: Koha/Plugin/EDS/bootstrap/modules/eds-detail.tt [hunk 1]
```diff
@@ -1287,7 +1287,7 @@
 [% END %]
 [% IF ( OpacStarRatings != 'disable' ) %]<script type="text/javascript" src="[% interface %]/[% theme %]/lib/jquery/plugins/jquery.rating.js"></script>[% END %]
 
-[% IF ( OpacHighlightedWords ) %]<script type="text/javascript" src="[% interface %]/[% theme %]/lib/jquery/plugins/jquery.highlight-3.js"></script>[% END %]
+[% IF ( OpacHighlightedWords ) %]<script type="text/javascript" src="[% interface %]/[% theme %]/lib/jquery/plugins/jquery.highlight-5.js"></script>[% END %]
```
Comment (cwolfebsco) [inline Koha/Plugin/EDS/bootstrap/modules/eds-detail.tt:1290]: This line should be reviewed, please. Both on http://koha-2311.ebsco-gss.net/ and my test environment (VM) this file is shown as 404. Example from ebsco-gss.net

Request URL:
http://koha-2311.ebsco-gss.net/opac-tmpl/bootstrap/lib/jquery/plugins/jquery.highlight-5.js

If you look on the same site with a catalog search the file is given as:
http://koha-2311.ebsco-gss.net/opac-tmpl/lib/jquery/plugins/jquery.highlight-5_23.1105000.js

Example of inclusion of that file
https://github.com/Koha-Community/Koha/blob/23.11.x/koha-tmpl/opac-tmpl/bootstrap/en/modules/opac-results.tt

```
[% IF ( OpacHighlightedWords ) %]
        [% Asset.js("lib/jquery/plugins/jquery.highlight-5.js") | $raw %]
[% END %]
```
    
  Obviously I am not a KOHA expert, I just noticed this 404 so do with it, as you please

File: Koha/Plugin/EDS/bootstrap/modules/eds-results.tt [hunk 2]
```diff
@@ -483,7 +483,7 @@
     [% Asset.js("js/authtoresults.js") | $raw %]
     [% Asset.js("lib/hc-sticky.js") | $raw %]
     [% IF ( OpacHighlightedWords ) %]
-        [% Asset.js("lib/jquery/plugins/jquery.highlight-3.js") | $raw %]
+        [% Asset.js("lib/jquery/plugins/jquery.highlight-5.js") | $raw %]
```
Comment (cwolfebsco) [inline Koha/Plugin/EDS/bootstrap/modules/eds-results.tt:486]: Same comment as for eds-detail.tt

File: Koha/Plugin/EDS/bootstrap/modules/pfi-results.tt [hunk 3]
```diff
@@ -412,7 +412,7 @@
 [% IF ( OpacStarRatings == 'all' || Koha.Preference('Babeltheque') ) %]<script type="text/javascript" src="[% interface %]/[% theme %]/lib/jquery/plugins/jquery.rating.js"></script>[% END %]
 [% IF ( OverDriveEnabled ) %]<script type="text/javascript" src="[% interface %]/[% theme %]/js/overdrive.js"></script>[% END %]
 <script type="text/javascript" src="[% interface %]/[% theme %]/lib/jquery/plugins/jquery.checkboxes.min.js"></script>
-[% IF ( OpacHighlightedWords ) %]<script type="text/javascript" src="[% interface %]/[% theme %]/lib/jquery/plugins/jquery.highlight-3.js"></script>
+[% IF ( OpacHighlightedWords ) %]<script type="text/javascript" src="[% interface %]/[% theme %]/lib/jquery/plugins/jquery.highlight-5.js"></script>
```
Comment (cwolfebsco) [inline Koha/Plugin/EDS/bootstrap/modules/pfi-results.tt:415]: same as eds-detail.tt comment

File: version.json [hunk 4]
```diff
@@ -1,10 +1,18 @@
...
+        {
+            "author": "omuzzy@ebsco.com",
+            "note": "fixes issue where EDS items cannot be downloaded from cart"
+        },
+        {
+            "author": "omuzzy@ebsco.com",
+            "note": "fixes other minor 23.11 errors"
         }
     ],
     "ShaAdd": "https://widgets.ebscohost.com/prod/api/koha/sha/1711.json"
```
Comment (cwolfebsco) [inline version.json:18]: Super ignorant question - this file references 1711.json. If you call upon the file it responds with:

```
{
    "edsplugin": {
        "version": [
            {
                "number": "17.11000",
                "sha": "2c43b43ab905685632f14cc38c6d85e174a60833"
            }
        ]
    }
}
```

which also references 1711. It doesn't seem to cause headaches, but I am curious...

File: Koha/Plugin/EDS/bootstrap/modules/eds-detail.tt [hunk 5]
```diff
@@ -1287,7 +1287,7 @@
 [% END %]
 [% IF ( OpacStarRatings != 'disable' ) %]<script type="text/javascript" src="[% interface %]/[% theme %]/lib/jquery/plugins/jquery.rating.js"></script>[% END %]
```
Comment (cwolfebsco) [inline Koha/Plugin/EDS/bootstrap/modules/eds-detail.tt:1288]: Both on my test environment and ebsco-gss.net the jquery.rating.js and jquery.rating.css seems to fail.

Request URL:
http://koha-2311.ebsco-gss.net/opac-tmpl/bootstrap/lib/jquery/plugins/jquery.rating.js

When looking for OpacStarRatings, I only came up with usage such as this:

```
    [% IF ( Koha.Preference('OpacStarRatings') == 'all' ) %]
        [% Asset.js("lib/jquery/plugins/jquery.barrating.min.js") | $raw %]
        [% Asset.js("js/ratings.js") | $raw %]
    [% END %]
```
Looking at relevant directory https://github.com/Koha-Community/Koha/tree/23.11.x/koha-tmpl/opac-tmpl/lib/jquery/plugins also doesn't show that jquery.rating.js - something that while not touched in this PR, we probably should investigate

File: Koha/Plugin/EDS/opac/2311/opac-downloadcart.pl
Comment (cwolfebsco) [inline Koha/Plugin/EDS/opac/2311/opac-downloadcart.pl:1]: I successfully added three records to my cart and downloaded as RIS

**Review by cwolfebsco — CHANGES_REQUESTED** [review]
I've tested on http://koha-2311.ebsco-gss.net and my local machine. The functionality seems to work (congrats!) but I found a few questions that seem unrelated, but still worth a quick investigation.

I also noted an error suggesting that is_logged_in is declared twice, but I was unable to find where that happens.

Uncaught SyntaxError: Identifier 'is_logged_in' has already been declared (at eds-search.pl?q=Search?query-1=AND,:{otzberg}&default=1:40376:9)

````
<script>
    var Koha = {};
    function _(s) { return s } // dummy function for gettext
    const is_logged_in = false
</script>
```

File: version.json (hunk 4 above)
Comment (mmabrahamson) [reply version.json:18]: This has been used for many years as the sha for download verification.  We'll make a note to see if this is still best practice, or if there's a better way, with plans to address in a future PR.

File: Koha/Plugin/EDS/bootstrap/modules/eds-results.tt (hunk 2 above)
Comment (omuzzyEBSCO) [reply Koha/Plugin/EDS/bootstrap/modules/eds-results.tt:486]: This one was already written correctly.

File: Koha/Plugin/EDS/bootstrap/modules/pfi-results.tt (hunk 3 above)
Comment (omuzzyEBSCO) [reply Koha/Plugin/EDS/bootstrap/modules/pfi-results.tt]: Fixed in most recent commit: [68f871c](https://github.com/ebsco/edsapi-koha-plugin/pull/138/commits/68f871c213c34349431975a391e1d45d1c0dce78)
Can you please reinstall/retest? Thank you.

File: Koha/Plugin/EDS/bootstrap/modules/eds-detail.tt (hunk 5 above)
Comment (omuzzyEBSCO) [reply Koha/Plugin/EDS/bootstrap/modules/eds-detail.tt:1289]: Thank you for pointing this out, we are adding this to work planning for next PR.

File: Koha/Plugin/EDS/bootstrap/modules/eds-detail.tt (hunk 1 above)
Comment (cwolfebsco) [reply Koha/Plugin/EDS/bootstrap/modules/eds-detail.tt]: confirmed

**Comment by cwolfebsco** [comment]
Looks good - only found one more. Downloaded RIS File contains a URL back to EDS, which is missing ampersands, e.g.:

UR  - https://search.ebscohost.com/login.aspx?direct=trueandsite=eds-liveanddb=bsuandAN=76633744

Notice how those ampersands (&) were replaced with the word and. This might be a koha problem, but it will impact things a little at our end.

**Comment by omuzzyEBSCO** [comment]
> Looks good - only found one more. Downloaded RIS File contains a URL back to EDS, which is missing ampersands, e.g.:
> 
> UR - https://search.ebscohost.com/login.aspx?direct=trueandsite=eds-liveanddb=bsuandAN=76633744
> 
> Notice how those ampersands (&) were replaced with the word and. This might be a koha problem, but it will impact things a little at our end.

Updated in this commit: [e3c0713](https://github.com/ebsco/edsapi-koha-plugin/pull/138/commits/e3c07139bdf98ef3120306ce895d4a30384400af)
Can you please reinstall and test again? Thank you.

**Comment by cwolfebsco** [comment]
@omuzzyEBSCO  I had trouble with my test environment, so I ended up testing on http://koha-2311.ebsco-gss.net/. Things work well there. I did see something that again might not be a "you" problem, but I wanted to document so you at least had a chance to decide yourself.

Diacritic and other unicode characters can lead to unfavorable display, as in this example:
Screenshot 2024-12-12 at 07 14 10

This is the record:
http://koha-2311.ebsco-gss.net/plugin/Koha/Plugin/EDS/opac/eds-detail.pl?q=Retrieve?an=39622108|dbid=eoah

This doesn't seem to be a show-stopper type of problem, but one to be aware of, as it might be a future issue.

**Review by mmabrahamson — APPROVED** [review]
Meets all requirements for feature release.

**Pull request merged**